pennylane/grouping/group_observables.py

```diff
@@ -0,0 +1,258 @@
+                    list(GRAPH_COLOURING_METHODS.keys()), graph_colourer
+                )
+            )
+
+        self.graph_colourer = GRAPH_COLOURING_METHODS[graph_colourer.lower()]
+        self.observables = observables
+        self.__wire_map = None
+        self.__n_qubits = None
+        self.binary_observables = None
+        self.adj_matrix = None
+        self.grouped_paulis = None
+
+    def obtain_binary_repr(self, n_qubits=None, wire_map=None):
+        """Converts the list of Pauli words to a binary matrix.
+
+        Keyword args:
+            n_qubits (int): number of qubits to specify dimension of binary vector representation.
+            wire_map (dict): dictionary containing all wire labels used in the Pauli word as keys,
+                and unique integer labels as their values
+
+        Returns:
+            array[bool]: a column matrix of the Pauli words in binary vector representation.
+
+        """
+
+        if wire_map is None:
+            self.__wire_map = {
+                wire: c
+                for c, wire in enumerate(
+                    Wires.all_wires([obs.wires for obs in self.observables]).tolist()
+                )
+            }
+
+        else:
+            self.__wire_map = wire_map
+
+        self.binary_observables = convert_observables_to_binary_matrix(
+            self.observables, n_qubits, self.__wire_map
+        )
+
+        self.__n_qubits = n_qubits
+
+        return self.binary_observables
+
+    def construct_complement_adj_matrix_for_operator(self):
+        """Constructs the adjacency matrix for complement of Pauli graph.
+
+        The adjacency matrix for an undirected graph of N vertices is an N by N symmetric binary
+        matrix, where matrix elements of 1 denote an edge, and matrix elements of 0 denote no edge.
+
+        Returns:
+            array[bool]: the square and symmetric adjacency matrix.
+
+        """
+
+        if self.binary_observables is None:
+            self.obtain_binary_repr()
+
+        n_qubits = int(np.shape(self.binary_observables)[0] / 2)
+
+        if self.grouping_type == "qwc":
+            adj = get_qwc_complement_adj_matrix(self.binary_observables)
+
+        elif self.grouping_type in frozenset(["commuting", "anticommuting"]):
+            symplectic_form = np.block(
+                [
+                    [np.zeros((n_qubits, n_qubits)), np.eye(n_qubits)],
+                    [np.eye(n_qubits), np.zeros((n_qubits, n_qubits))],
+                ]
+            )
+            mat_prod = (
+                np.transpose(self.binary_observables) @ symplectic_form @ self.binary_observables
+            )
+
+            if self.grouping_type == "commuting":
+
+                adj = mat_prod % 2
+
+            elif self.grouping_type == "anticommuting":
+
+                adj = (mat_prod + 1) % 2
+                np.fill_diagonal(adj, 0)
+
+        self.adj_matrix = adj
+        return adj
```

Comment:
Would suggest a change similar to the one for `obtain_binary_repr`. Will help with removing the side effect of setting the instance attribute `adj_matrix` here.